instrumentation/elastic-opentelemetry-instrumentation-openai/tests/test_beta_chat_completions.py

```diff
@@ -0,0 +1,1544 @@
+            "tool_call_id": previous_message["tool_calls"][1]["id"],
+        },
+    ]
+
+    second_response = client.beta.chat.completions.parse(model=TEST_CHAT_MODEL, messages=messages + followup_messages)
+
+    spans = trace_exporter.get_finished_spans()
+    assert len(spans) == 2
+
+    first_span, second_span = spans
+    assert first_span.name == f"chat {TEST_CHAT_MODEL}"
+    assert first_span.kind == SpanKind.CLIENT
+    assert first_span.status.status_code == StatusCode.UNSET
+
+    address, port = address_and_port(client)
+    assert dict(first_span.attributes) == {
+        GEN_AI_OPERATION_NAME: "chat",
+        GEN_AI_REQUEST_MODEL: TEST_CHAT_MODEL,
+        GEN_AI_SYSTEM: "openai",
+        GEN_AI_RESPONSE_ID: first_response.id,
+        GEN_AI_RESPONSE_MODEL: TEST_CHAT_RESPONSE_MODEL,
+        GEN_AI_RESPONSE_FINISH_REASONS: ("tool_calls",),
+        GEN_AI_USAGE_INPUT_TOKENS: first_response.usage.prompt_tokens,
+        GEN_AI_USAGE_OUTPUT_TOKENS: first_response.usage.completion_tokens,
+        SERVER_ADDRESS: address,
+        SERVER_PORT: port,
+    }
+
+    assert second_span.name == f"chat {TEST_CHAT_MODEL}"
+    assert second_span.kind == SpanKind.CLIENT
+    assert second_span.status.status_code == StatusCode.UNSET
+
+    assert dict(second_span.attributes) == {
+        GEN_AI_OPERATION_NAME: "chat",
+        GEN_AI_REQUEST_MODEL: TEST_CHAT_MODEL,
+        GEN_AI_SYSTEM: "openai",
+        GEN_AI_RESPONSE_ID: second_response.id,
+        GEN_AI_RESPONSE_MODEL: TEST_CHAT_RESPONSE_MODEL,
+        GEN_AI_RESPONSE_FINISH_REASONS: ("stop",),
+        GEN_AI_USAGE_INPUT_TOKENS: second_response.usage.prompt_tokens,
+        GEN_AI_USAGE_OUTPUT_TOKENS: second_response.usage.completion_tokens,
+        SERVER_ADDRESS: address,
+        SERVER_PORT: port,
+    }
+
+    logs = logs_exporter.get_finished_logs()
+    assert len(logs) == 9
+    log_records = logrecords_from_logs(logs)
+
+    # first call events
+    system_message, user_message, choice = log_records[:3]
+    assert system_message.attributes == {"gen_ai.system": "openai", "event.name": "gen_ai.system.message"}
+    assert system_message.body == {"content": "You are a helpful assistant providing weather updates."}
+    assert user_message.attributes == {"gen_ai.system": "openai", "event.name": "gen_ai.user.message"}
+    assert user_message.body == {"content": "What is the weather in New York City and London?"}
+
+    assert_tool_call_log_record(
+        choice,
+        [
+            ToolCall(
+                id=previous_message["tool_calls"][0]["id"],
+                name="get_weather",
+                arguments_json='{"location": "New York City"}',
+            ),
+            ToolCall(
+                id=previous_message["tool_calls"][1]["id"], name="get_weather", arguments_json='{"location": "London"}'
+            ),
+        ],
+    )
+
+    # second call events
+    tool_call_telemetry = previous_message["tool_calls"]
```

Comment:
This is the only diff from `test_chat`. Though there it's also sort of a coincidence that semconv matches the API response, while in this case it doesn't since there's additional fields.